src/WebJobs.Script.WebHost/Security/KeyManagement/SimpleKubernetesClient.cs

```diff
@@ -0,0 +1,307 @@
+            }
+
+            var @namespace = await FileUtility.ReadAsync(NamespaceFile);
+            var url = $"{_environment.GetKubernetesApiServerUrl()}/api/v1/";
+            if (watchUrl)
+            {
+                url += "watch/";
+            }
+            url += $"namespaces/{@namespace}/{objectName}";
+            return (url, isSecret);
+        }
+
+        private async Task<HttpRequestMessage> GetRequest(HttpMethod method, string url, object payload = null, string contentType = null, bool closeConnection = true)
+        {
+            var token = await FileUtility.ReadAsync(TokenFile);
+            const string jsonContentType = "application/json";
+            contentType = contentType ?? jsonContentType;
+
+            var request = new HttpRequestMessage(method, url);
+            request.Headers.Add("Authorization", $"Bearer {token}");
+            request.Headers.Add("Accept", jsonContentType);
+
+            if (closeConnection)
+            {
+                request.Headers.Add("Connection", "close");
+            }
+
+            if (payload != null)
+            {
+                var jsonPayload = JsonConvert.SerializeObject(payload);
+                request.Content = new StringContent(jsonPayload, Encoding.UTF8, contentType);
+            }
+
+            return request;
+        }
+
+        private static async Task<IDictionary<string, string>> GetFromFiles(string path)
+        {
+            var files = await FileUtility.GetFilesAsync(path, "*");
+            var secrets = new Dictionary<string, string>(files.Length);
+            foreach (var file in files)
+            {
+                secrets.Add(Path.GetFileName(file), await FileUtility.ReadAsync(file));
+            }
+            return secrets;
+        }
+
+        private static HttpClient CreateHttpClient()
+        {
+            var client = new HttpClient(new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = ServerCertificateValidationCallback
+            });
+
+            client.DefaultRequestHeaders.Add("User-Agent", ScriptConstants.HostUserAgent);
+            return client;
+        }
+
+        private static bool ServerCertificateValidationCallback(
+            HttpRequestMessage request,
+            X509Certificate2 certificate,
+            X509Chain certChain,
+            SslPolicyErrors sslPolicyErrors)
+        {
+            if (sslPolicyErrors == SslPolicyErrors.None)
+            {
+                // certificate is already valid
+                return true;
+            }
+            else if (sslPolicyErrors == SslPolicyErrors.RemoteCertificateNameMismatch ||
+                sslPolicyErrors == SslPolicyErrors.RemoteCertificateNotAvailable)
+            {
+                // api-server cert must exist and have the right subject
+                return false;
+            }
+
+            // only remaining error state is RemoteCertificateChainErrors
```

Comment:
nit: To safeguard against potential (even if unlikely) changes to the enum, it would be good to still put this behind a check for the (currently) last remaining enumeration value here.